### PR TITLE
Enabling ipFamilies and ipFamilyPolicy

### PR DESCRIPTION
**Pull request by yuriolisa**
**Description:** <Describe what has changed.>

Enabling ipFamilies and ipFamilyPolicy to be configured via OpenTelemetryCollector
**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #2958 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
looks like e2e is failing here

**Review by pavolloffay — APPROVED** [review]

**Review by swiatekm — COMMENTED** [review]
Are these e2e tests checking anything a controller test wouldn't? I'd rather have that if you aren't going to check if ipv6 actually works. We could even add a controller test that sets all the optional attributes in OpenTelemetryCommonFields.

Other than that, LGTM.

**Comment by yuriolisa** [comment]
> Are these e2e tests checking anything a controller test wouldn't? I'd rather have that if you aren't going to check if ipv6 actually works. We could even add a controller test that sets all the optional attributes in OpenTelemetryCommonFields.
> 
> Other than that, LGTM.

Great idea, @swiatekm. I add the tests on both sides, controller and e2e.

**Comment by pavolloffay** [comment]
e2e tests failed

**Review by jaronoff97 — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -496,7 +496,7 @@ operator-sdk: $(LOCALBIN)
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: generate-bundle
 generate-bundle: kustomize operator-sdk manifests set-image-controller api-docs
-	sed -i 's/minKubeVersion: .*/minKubeVersion: $(MIN_KUBERNETES_VERSION)/' config/manifests/$(BUNDLE_VARIANT)/bases/opentelemetry-operator.clusterserviceversion.yaml
+	sed -e 's/minKubeVersion: .*/minKubeVersion: $(MIN_KUBERNETES_VERSION)/' config/manifests/$(BUNDLE_VARIANT)/bases/opentelemetry-operator.clusterserviceversion.yaml
```
Comment (swiatekm) [inline Makefile:499]: Why are we making this change?

**Review by swiatekm — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (yuriolisa) [reply Makefile:499]: In certain MacOs/Linux distributions the `sed -i` doesn't work.

**Pull request merged**